src/JsonSchema/Constraints/Factory.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the JsonSchema package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JsonSchema\Constraints;
+
+use JsonSchema\Exception\InvalidArgumentException;
+use JsonSchema\Uri\UriRetriever;
+use JsonSchema\Validator;
+
+/**
+ * Factory for centralize constraint initialization.
+ */
+class Factory
```

Comment:
This library appears to duplicate the namespace in the class name. 

For example:
``` JsonSchema\Uri\UriRetriever``` instead of ``` JsonSchema\Uri\Retriever```
```JsonSchema\Constraints\NumberConstraint``` instead of ```JsonSchema\Constraints\Number```

This should maybe be called ```ConstraintFactory```